hazelcast/src/main/java/com/hazelcast/map/impl/eviction/EvictionChecker.java

```diff
@@ -119,8 +120,15 @@ public double calculatePerNodeMaxRecordStoreSize(RecordStore recordStore) {
         int memberCount = nodeEngine.getClusterService().getSize(DATA_MEMBER_SELECTOR);
         int partitionCount = nodeEngine.getPartitionService().getPartitionCount();
 
-        return (1D * configuredMaxSize * memberCount / partitionCount);
-
+        final double perNodeMaxRecordStoreSize = (1D * configuredMaxSize * memberCount / partitionCount);
+        if (perNodeMaxRecordStoreSize < 1 && !misconfiguredPerNodeMaxSizeWarningLogged) {
```

Comment:
Can an `AtomicBoolean` fit better to prevent possible multiple similar logs?

`if (perNodeMaxRecordStoreSize < 1 && misconfiguredPerNodeMaxSizeWarningLogged.compareAndSet(false,true))`